pkg/util/detector/detector.go

```diff
@@ -1013,24 +1012,8 @@ func (d *ResourceDetector) OnResourceBindingUpdate(_, newObj interface{}) {
 
 // OnClusterResourceBindingDelete handles object delete event.
 func (d *ResourceDetector) OnClusterResourceBindingDelete(obj interface{}) {
-	unstructuredObj, ok := obj.(*unstructured.Unstructured)
-	if !ok {
-		klog.Errorf("Invalid object type: %v", reflect.TypeOf(obj))
-		return
-	}
-
-	binding := &workv1alpha1.ClusterResourceBinding{}
-	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredObj.UnstructuredContent(), binding); err != nil {
-		klog.Errorf("Failed to convert unstructured to typed object: %v", err)
-		return
-	}
-
-	objRef := binding.Spec.Resource
-	err := d.CleanupResourceTemplateStatus(binding.Spec.Resource)
-	if err != nil {
-		// Just log when error happened as there is no queue for delete event.
-		klog.Warningf("Failed to cleanup resource(kind=%s, %s/%s) status: %v", objRef.Kind, objRef.Namespace, objRef.Name, err)
-	}
+	// The life cycle of ClusterResourceBinding is controlled by resource templates,
+	// so when ClusterResourceBinding has been deleted, it's no need to clean up resource status.
```

Comment:
Not exact enough.
But why not remove this method if we don't care about the deletion event?